openmc/source.py

```diff
@@ -349,8 +362,18 @@ class SourceParticle:
         Type of the particle
 
     """
-    def __init__(self, r=(0., 0., 0.), u=(0., 0., 1.), E=1.0e6, time=0.0, wgt=1.0,
-                 delayed_group=0, surf_id=0, particle=ParticleType.NEUTRON):
+    def __init__(
+        self,
+        r: Tuple[float, float, float] = (0., 0., 0.),
+        u: Tuple[float, float, float] = (0., 0., 1.),
```

Comment:
These don't necessary have to be tuples, any iterable will do
```suggestion
        r: typing.Iterable[float] = (0., 0., 0.),
        u: typing.Iterable[float] = (0., 0., 1.),
```

Reply:
I was just thinking with a tuple we can specify that it needs to be three entries long

Reply:
Yes, that would be nice to be able to specify the length, but then a static type checker would not accept the following:
```Python
openmc.SourceParticle([1., 2, 5.], [0., 0., 1.], ...)
```
which is indeed valid